### DISCUSSION TITLE
Is this library thread-safe?

### DISCUSSION THREAD

**Discussion by UbiquitousWhite** [discord-net/Discord.Net#2133] [Q&A]
I can not find any information on this. I want to run Cronjobs on my Discord server. Is this libary thread-safe?

**Comment by essoperagma** [comment]
I basically have the same question. 
My bot is performing some work that is expected to take longer than 3 seconds, so I'm offloading it to a different task not to block the client's even loop.

The issue is, when the task completes, I'm worried about accessing the DiscordSocketClient object. Is there an event queue that we can post our work to?

The message pump proposed here (https://github.com/discord-net/Discord.Net/issues/1913) would be sufficient for my needs. But I'm wondering if I can still perform simple tasks like sending a message and updating guild user nickname without worrying about synchronization.

**Reply by essoperagma** [reply]
@quinchs Can you comment on this? Is there a workaround I can utilize until a message pump is implemented?

**Reply by budgetdevv** [reply]
Pretty sure it is thread-safe

**Comment by BowserKingKoopa** [comment]
I'm looking for an answer to this as well. Is DiscordSocketClient thread safe?

**Reply by budgetdevv** [reply]
Yes
